app/main/utils.py

```diff
@@ -384,15 +384,89 @@ def _replicate_office_contacts(
     return replicated_contacts
 
 
-def get_office_tags(office: Office):
+def get_firm_tags(firm: Firm):
     tags: list[Tag] = []
-    if office.inactive_date:
+    if firm.inactive_date:
         tags.append(Tag(TagType.INACTIVE))
     return tags
 
 
-def get_firm_tags(firm: Firm):
+def get_office_tags(office: Office):
     tags: list[Tag] = []
-    if firm.inactive_date:
+    if office.inactive_date:
         tags.append(Tag(TagType.INACTIVE))
     return tags
+
+
+def get_firm_account_number(firm: Firm | int) -> str | None:
+    """Gets the account number for a given firm or firm_id.
+
+    A firm's account number is the firm_office_code of that firm's head office.
+
+    Args:
+        firm: Either a Firm instance or a firm_id (int)
+
+    Returns:
+        The firm's account number (firm_office_code), or None if no head office exists
+
+    Raises:
+        RuntimeError: If Provider Data API is not initialized
+        ValueError: If firm is neither a Firm instance nor an int
+    """
+    pda: MockProviderDataApi = current_app.extensions.get("pda")
```

Comment:
Do we need to add type hints for the Mock API, given that it will be replaced by the real API class once it's fully implemented?